DQM/BeamMonitor/plugins/BeamSpotProblemMonitor.cc

```diff
@@ -189,182 +150,156 @@ void BeamSpotProblemMonitor::analyze(const Event& iEvent,
     return;
   }
 
-    BeamSpotStatus_ = 0.;
-
-
-    // Checking TK status
-    Handle<DcsStatusCollection> dcsStatus;
-    iEvent.getByToken(dcsStatus_, dcsStatus);
-    for (int i=0;i<6;i++) dcsTk[i]=true;
-    for (DcsStatusCollection::const_iterator dcsStatusItr = dcsStatus->begin(); 
-         dcsStatusItr != dcsStatus->end(); ++dcsStatusItr) {
-      if (!dcsStatusItr->ready(DcsStatus::BPIX))   dcsTk[0]=false;
-      if (!dcsStatusItr->ready(DcsStatus::FPIX))   dcsTk[1]=false;
-      if (!dcsStatusItr->ready(DcsStatus::TIBTID)) dcsTk[2]=false;
-      if (!dcsStatusItr->ready(DcsStatus::TOB))    dcsTk[3]=false;
-      if (!dcsStatusItr->ready(DcsStatus::TECp))   dcsTk[4]=false;
-      if (!dcsStatusItr->ready(DcsStatus::TECm))   dcsTk[5]=false;
-    }  
-
-     bool AllTkOn = true;
-        for (int i=0; i<5; i++) 
-            {
-              if (!dcsTk[i]) {
-                              AllTkOn = false;
-                       
-                             }
-            }
-
-
-     //If tracker is ON and collision is going on then must be few track ther
-     edm::Handle<reco::TrackCollection> TrackCollection;
-     iEvent.getByToken(trkSrc_, TrackCollection);
-     const reco::TrackCollection *tracks = TrackCollection.product();
-     for ( reco::TrackCollection::const_iterator track = tracks->begin();track != tracks->end();++track ) 
-      {
-         if(track->pt() > 1.0)Ntracks_++;
-          if(Ntracks_> 200) break;
-      }
+  beamSpotStatus_ = 0.;
+
+
+  // Checking TK status
+  Handle<DcsStatusCollection> dcsStatus;
+  iEvent.getByToken(dcsStatus_, dcsStatus);
+  std::array<bool,6> dcsTk;
+  for (auto &e: dcsTk) { e=true; }
+  for (auto const& status: *dcsStatus ) {
+    if (!status.ready(DcsStatus::BPIX))   dcsTk[0]=false;
+    if (!status.ready(DcsStatus::FPIX))   dcsTk[1]=false;
+    if (!status.ready(DcsStatus::TIBTID)) dcsTk[2]=false;
+    if (!status.ready(DcsStatus::TOB))    dcsTk[3]=false;
+    if (!status.ready(DcsStatus::TECp))   dcsTk[4]=false;
+    if (!status.ready(DcsStatus::TECm))   dcsTk[5]=false;
+  }
+
+  bool allTkOn = true;
+  //BUG??? DcsStatus::TECm is ignored
+  for (int i=0; i<5; i++) {
+    if (!dcsTk[i]) {
+      allTkOn = false;
+    }
+  }
+
+
+  //If tracker is ON and collision is going on then must be few track ther
+  edm::Handle<reco::TrackCollection> trackCollection;
+  iEvent.getByToken(trkSrc_, trackCollection);
+  for ( auto const&  track : *trackCollection) {
+    if(track.pt() > 1.0)nTracks_++;
+    if(nTracks_> 200) break;
+  }
 
 
 
   // get scalar collection and BeamSpot
   Handle<BeamSpotOnlineCollection> handleScaler;
   iEvent.getByToken( scalertag_, handleScaler);
-     
-   // beam spot scalar object
-   BeamSpotOnline spotOnline;
-     
-   bool fallBackToDB=false;
-        ALARM_ON_  = false;
-
- 
-   if (!handleScaler->empty())
-     {
-      spotOnline = * ( handleScaler->begin() );
+
+  // beam spot scalar object
+  BeamSpotOnline spotOnline;
+
+  bool fallBackToDB=false;
```

Comment:
I find it odd that if `handleScaler->empty()` is `true` then this code does not fall back to the DB as it will do if it thinks the fit result is not valid. Is that the actual intended behavior?